### ISSUE TITLE
CSS inlined SVG Issue

### ISSUE THREAD

**Issue by bytesbrigade** [django-compressor/django-compressor#544]
If you have inline svg data in you css file like:

``` css
background-image: url("data:image/svg+xml;charset=utf-8,<svg xmlns='http://www.w3.org/2000/svg'
```

and you compress the file with django compressor the double quotes are changed to single quotes. 

``` css
background-image: url('data:image/svg+xml;charset=utf-8,<svg xmlns='http://www.w3.org/2000/svg' 
```

Which cause that the svg is ignored by the browser.

**Comment by gmeans** [comment]
I think we experienced the same issue with a data-uri reference as well:

background-image: data-uri('image/svg+xml;charset=UTF-8', '../images/hexagon.svg');

After compression it looks like compressor found the file but wrote it incorrectly into the compressed stylesheet causing the browser to ignore it.

**Comment by wwform** [comment]
+1

**Comment by diox** [comment]
It should be fixed in django-compressor 2.1, which was released less than 2 weeks ago. See https://github.com/django-compressor/django-compressor/pull/762

**Comment by wwform** [comment]
Just noticed and updated, sorry. There still seems to be an issue in 2.1 that's converting _one_ of my single quotes inside inlined svgs into double quotes, thus breaking the svg.

This:
`background-image: url("data:image/svg+xml;charset=utf8,%3Csvg width='1000' height='1000' viewBox='0 0 1000 1000' xmlns='http://www.w3.org/2000/svg'%3E%3Cdefs%3E%3ClinearGradient x1='50%' y1='0%' x2='50%' y2='100%' id='gdnt'%3E%3Cstop stop-color='%23e83f39' offset='0%'%3E%3C/stop%3E%3Cstop stop-color='%232cabe2' offset='100%'%3E%3C/stop%3E%3C/linearGradient%3E%3C/defs%3E%3Cg fill='url(%23gdnt)'%3E%3Crect x='0' y='0' width='1000' height='1000'%3E%3C/rect%3E%3C/g%3E%3C/svg%3E");`

Becomes:
`background-image: url("data:image/svg+xml;charset=utf8,%3Csvg width='1000' height='1000' viewBox='0 0 1000 1000' xmlns='http://www.w3.org/2000/svg'%3E%3Cdefs%3E%3ClinearGradient x1='50%' y1='0%' x2='50%' y2='100%' id='gdnt'%3E%3Cstop stop-color='%23e83f39' offset='0%'%3E%3C/stop%3E%3Cstop stop-color='%232cabe2' offset='100%'%3E%3C/stop%3E%3C/linearGradient%3E%3C/defs%3E%3Cg fill='url(%23gdnt")'%3E%3Crect x='0' y='0' width='1000' height='1000'%3E%3C/rect%3E%3C/g%3E%3C/svg%3E");`

**Comment by diox** [comment]
:(

**Comment by wwform** [comment]
This is probably caused by some regex looking for `url(` -- and as my data url was only partially urlendcoded (notably not the brackets), part of my svg code matched. Once I encode the brackets too, the additional quotation mark isn't inserted anymore. Probably worth pointing that out in the docs (or somewhere).

**Comment by gertsteyn** [comment]
Getting the same issue, but when compressing bootstrap 4 alpha 6:

`background-image: url("data:image/svg+xml;charset=utf8,%3Csvg viewBox='0 0 32 32' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath stroke='rgba(0, 0, 0, 0.6)' stroke-width='2' stroke-linecap='round' stroke-miterlimit='10' d='M4 8h24M4 16h24M4 24h24'/%3E%3C/svg%3E"); }
`

ends up as

`background-image: url("data:image/svg+xml;charset=utf8,%3Csvg viewBox='0 0 32 32' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath stroke='rgba(0, 0, 0, 0.6")' stroke-width='2' stroke-linecap='round' stroke-miterlimit='10' d='M4 8h24M4 16h24M4 24h24'/%3E%3C/svg%3E"); }
`

The " before the ) breaks the entire stylesheet, chrome stops processing the CSS entirely when it gets there... is this a bootstrap or django-compressor issue?

**Comment by karyon** [comment]
the remaining issues look like a duplicate of #764 and should be fixed with #828.
